my-code/src/Contexts/MovieDetails.tsx

```diff
@@ -0,0 +1,55 @@
+import axios from 'axios';
+import { createContext, ReactNode, useContext, useState } from 'react';
+
+interface MovieDetailsProviderProps {
+    children: ReactNode
+}
+
+interface MovieDetailsContextProps {
+    getMovieInfo: (imdbID: string) => Promise<void>
+    movieInfo: MovieInfoProps
+    isLoading: boolean
+}
+
+interface MovieInfoProps {
+    imdbID: string
+    Runtime: string
+    Year: string
+    Rated: string
+    Title: string
+    Ratings: [{ Source: string, Value: string }]
+    Plot: string
+    Actors: string;
+    Genre: string;
+    Director: string
+    Poster: string
+}
+
+const MovieDetailsContext = createContext({} as MovieDetailsContextProps)
+
+export function MovieDetailsProvider({ children }: MovieDetailsProviderProps) {
+
+    const [movieInfo, setMovieInfo] = useState({} as MovieInfoProps)
```

Comment:
```suggestion
      const [movieInfo, setMovieInfo] = useState<MovieInfoProps>({});
```
Same thing here, casting can lead to errors